src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,568 @@
+
+        df_zonal_stats = pd.concat(df_list).reset_index(drop=True)
+        return df_zonal_stats
+
+
+@xr.register_dataset_accessor("aat")
+class AatRasterDataset(AatRasterMixin, RasterDataset):
+    """AA toolbox extension for xarray.Dataset."""
+
+    def __init__(self, xarray_object):
+        super().__init__(xarray_object)
+
+    def get_raster_array(self, var: str) -> xr.DataArray:
+        """Get xarray.DataArray from variable and keep dimensions.
+
+        Accessing a component xarray.DataArray using the
+        non-coordinate variable name loses and dimensions set
+        through ``rio`` or ``aat``. This includes ``x_dim``,
+        ``y_dim``, and ``t_dim`` that have to be specifically
+        set using ``rio.set_spatial_dims()`` or
+        ``aat.set_time_dim()`` respectively. For any dataset
+        ``ds``, ``ds.get_raster_array("var")`` will retrieve
+        the data array without losing the dimensions. Using
+        ``ds["var"]`` will lose the dimensions.
+
+        Parameters
+        ----------
+        var : str
+            Name of variable.
+
+        Returns
+        -------
+        xarray.DataArray
+            A data array.
+
+        Examples
+        --------
+        >>> import xarray
+        >>> import numpy
+        >>> temp = 15 + 8 * numpy.random.randn(4, 4, 3)
+        >>> precip = 10 * np.random.rand(4, 4, 3)
+        >>> ds = xarray.Dataset(
+        ...   {
+        ...     "temperature": (["lat", "lon", "F"], temp),
+        ...     "precipitation": (["lat", "lon", "F"], precip)
+        ...   },
+        ...   coords={
+        ...     "lat":numpy.array([87, 88, 89, 90]),
+        ...     "lon":numpy.array([5, 120, 199, 360]),
+        ...     "F": pd.date_range("2014-09-06", periods=3)
+        ...   }
+        ... )
+        >>> ds.aat.set_time_dim("F", inplace=True)
+        >>> da = ds.aat.get_raster_array("temperature")
+        >>> da.aat.t_dim
+        'F'
+        >>> # directly accessing array loses set dimensions
+        >>> ds['temperature'].aat.t_dim # doctest: +NORMALIZE_WHITESPACE
+        Traceback (most recent call last):
+            ...
+        rioxarray.exceptions.DimensionError: Time dimension not found.
+            'aat.set_time_dim()' or using 'rename()' to change the
+            dimension name to 't' can address this.
+        Data variable: temperature
+        """
+        obj = self._obj[var]
+        # rioxarray attributes
+        if self._x_dim is not None and self._y_dim is not None:
+            obj.rio.set_spatial_dims(
+                x_dim=self._x_dim, y_dim=self._y_dim, inplace=True
+            )
+        if self._crs is not None:
+            obj.rio.set_crs(self._crs, inplace=True)
+
+        # raster module attributes
+        if self._t_dim is not None:
+            obj.aat.set_time_dim(self._t_dim, inplace=True)
+
+        return obj
+
+    def compute_raster_stats(
+        self, vars: Union[List[str], None] = None, **kwargs: Any
```

Comment:
You shouldn't use the name `vars` because it's one of the [builtin functions](https://docs.python.org/3/library/functions.html)

Reply:
Oh damn good catch!!